apps/api/src/modules/notifications/notifications.service.ts

```diff
@@ -98,12 +104,65 @@ export class NotificationsService {
     });
   }
 
-  getAllNotifications(): Promise<Notification[]> {
-    this.logger.log('Getting all active notifications');
-    return this.notificationRepository.find({
-      where: {
-        status: Status.ACTIVE,
-      },
+  async getAllNotifications(options: QueryOptionsDto): Promise<NotificationResponse> {
+    this.logger.log('Getting all active notifications with options');
+
+    const whereConditions = { status: Status.ACTIVE };
+
+    options.filters?.forEach((filter) => {
+      const field = filter.field;
+      const value = filter.value;
+
+      switch (filter.operator) {
+        case 'eq':
+          whereConditions[field] = Equal(value);
+          break;
+        case 'ne':
+          whereConditions[field] = Not(Equal(value));
+          break;
+        case 'contains':
+          if (typeof value === 'string') {
+            whereConditions[field] = Like(`%${value}%`);
+          }
+
+          break;
+        case 'gt':
+          if (this.isDateField(field)) {
+            whereConditions[field] = MoreThan(new Date(value));
+          } else {
+            whereConditions[field] = MoreThan(value);
+          }
+
+          break;
+        case 'lt':
+          if (this.isDateField(field)) {
+            whereConditions[field] = LessThan(new Date(value));
+          } else {
+            whereConditions[field] = LessThan(value);
+          }
+
+          break;
+      }
     });
+
+    const queryOptions: FindManyOptions<Notification> = {
+      where: whereConditions,
+      skip: options.offset,
+      take: options.limit,
+      order: options.sortBy
+        ? { [options.sortBy]: options.sortOrder === SortOrder.ASC ? 'ASC' : 'DESC' }
```

Comment:
Can we use `SortOrder` itself for the values for `ASC` and `DESC` instead of the strings here (like `SortOrder.ASC`)?

Reply:
Yes it is suppose to be `SortOrder.ASC` an `SortOrder.DESC`. Will fix this.

Reply:
Updated in https://github.com/OsmosysSoftware/osmo-notify/pull/115/commits/9bf550bc9749e70b27bd1a3b6cdcaacc6b79e124